### ISSUE TITLE
"mayaUSD.mod" is not being filled with values.

### ISSUE THREAD

**Issue by KelSolaar** [Autodesk/maya-usd#21]
Hi,

I'm building the plugin with CMake on Windows and I see that the `mayaUSD.mod` file is not being updated, its content is actually the same than the source `mayaUSD.mod.template` file:

```
+ USD <USD_VERSION> <PATH_TO_PXR_USD_DIR>
PYTHONPATH+:=lib/python
PATH+:=bin
PATH+:=lib

+ PXR_USDMAYA <USD_VERSION> <PATH_TO_PXR_USDMAYA_DIR>
PYTHONPATH+:=lib/python
PATH+:=maya/lib
XBMLANGPATH+:=maya/lib/usd/usdMaya/resources
MAYA_SCRIPT_PATH+:=maya/lib/usd/usdMaya/resources
MAYA_PLUG_IN_PATH+:=maya/plugin
PXR_PLUGINPATH_NAME+:=lib/usd

+ AL_USDMaya <AL_MAYAUSD_VER> <PATH_TO_AL_USDMaya_DIR>
PYTHONPATH+:=lib/python
PATH+:=lib
MAYA_PLUG_IN_PATH+:=plugin
PXR_PLUGINPATH_NAME+:=lib/usd
```

It seems that it should be filled by the build process as I can see a relevant call to `configure_file` at the very end of the `CMakeLists.txt` file but it does not seem to work as expected.

Cheers,

Thomas

**Comment by HamedSabri-adsk** [comment]
Hi KelSolaar, 
You are correct.  We are not currently filling out those variable placeholders. Yes I agree with you that the cmake should handle populating them since we already know were the paths are.

**Comment by chunghob** [comment]
Any luck on what should be configured for the maya.env? I have configured it similar to how the [USD docs say to setup pathing](https://graphics.pixar.com/usd/docs/Maya-USD-Plugins.html), and am able to see the plug-in mlls in the manager, but they still don't load when I enable them. 
I have hard coded all the paths and the error I get is: `The specified module could not be found.`

We have some specific env setup for our pipeline and it would be great if there was a setup doc or section in the build.md for loading/configuring in maya. 

Side note is there a user group similar to USD and AL_USD for maya-usd?

**Comment by KelSolaar** [comment]
@HamedSabri-adsk : I have for now patched the files with our build system.

@chunghob : Could it be related to #23 ? For what it is worth my mod file is as follows:

```
+ PXR_USDMAYA 19.05 ./plugin/pxr
PYTHONPATH+:=lib/python
PATH+:=maya/lib
XBMLANGPATH+:=maya/lib/usd/usdMaya/resources
MAYA_SCRIPT_PATH+:=maya/lib/usd/usdMaya/resources
MAYA_PLUG_IN_PATH+:=maya/plugin
PXR_PLUGINPATH_NAME+:=lib/usd

+ AL_USDMaya 0.31.1 ./plugin/al
PYTHONPATH+:=lib/python
PATH+:=lib
MAYA_PLUG_IN_PATH+:=plugin
PXR_PLUGINPATH_NAME+:=lib/usd
```

As per #23, I'm ensuring that the _maya-usd_ `PYTHONPATH` entries appear before those of _USD_.

**Comment by HamedSabri-adsk** [comment]
@KelSolaar 
Your Mod looks good to me. It is totally fine to use relative paths.
The versions that you have set are fine too.  To add to mod dramas, you could have put any numbers that you like for those versions ( e.g 0.0.0 ).  It is just a place holder.

**Comment by rhaleblian** [comment]
I consulted CMake documentation on `configure_file` to amend the template to use CMake variables and the result is close to working for me. Example:

```
+ USD @USD_VERSION@ @PXR_USD_LOCATION@
PYTHONPATH+:=lib/python
PATH+:=bin
PATH+:=lib

+ PXR_USDMAYA @PXR_MAJOR_VERSION@.@PXR_MINOR_VERSION@.@PXR_PATCH_VERSION@ @CMAKE_INSTALL_PREFIX@/plugin/pxr
PYTHONPATH+:=lib/python
PATH+:=maya/lib
XBMLANGPATH+:=maya/lib/usd/usdMaya/resources
MAYA_SCRIPT_PATH+:=maya/lib/usd/usdMaya/resources
MAYA_PLUG_IN_PATH+:=maya/plugin

+ MayaUSD_LIB @MAYAUSD_VERSION@ @CMAKE_INSTALL_PREFIX@
PATH+:=lib
PXR_PLUGINPATH_NAME+:=lib/usd
MAYA_WANT_UFE_SELECTION=1
VP2_RENDER_DELEGATE_PROXY=1

+ MayaUSD @MAYAUSD_VERSION@ @CMAKE_INSTALL_PREFIX@/plugin/adsk
MAYA_PLUG_IN_PATH+:=plugin
MAYA_SCRIPT_PATH+:=scripts

+ AL_USDMaya @MAYAUSD_VERSION@ @CMAKE_INSTALL_PREFIX@/plugin/al
PYTHONPATH+:=lib/python
PATH+:=lib
MAYA_PLUG_IN_PATH+:=plugin
PXR_PLUGINPATH_NAME+:=lib/usd
PXR_PLUGINPATH_NAME+:=plugin

+ MTOH @MAYAUSD_VERSION@ @CMAKE_INSTALL_PREFIX@
MAYA_PLUG_IN_PATH+:=lib/maya
```
As discussed in another issue, i prefer not having USD specifications in here but can't take that out without breakage ATM.

**Comment by rhaleblian** [comment]
I could provide a PR, but some cross-company CLA issues need to be cleared up first.

**Comment by mattyjams** [comment]
I just submitted PR #29 for this before stumbling onto the discussion here. For now, it's against the dev branch, but as @rhaleblian suggests it could easily be amended for refactoring_sandbox. Let me know also if people prefer the at-style syntax to curly braces.

**Comment by BigRoy** [comment]
This issue seems to have been resolved with #29 - can someone close this?
